packages/@react-types/shared/src/collections.d.ts

```diff
@@ -92,12 +92,13 @@ export interface SortDescriptor {
 
 export type SortDirection = 'ascending' | 'descending';
 
-export interface KeyboardDelegate {
+export interface KeyboardDelegate<T> {
+  // TODO: what do we think about this update to the keyboard delegate api?
```

Comment:
As mentioned previously, changes in this PR include a update to the options provided to KeyboardDelegate's `getKey` getters

Reply:
Try doing `T extends object` instead to make this a non-breaking change for someone who made their own keyboard delegate